### PR TITLE
feat: add Dockerfile for ocb

### PR DESCRIPTION
**Pull request by sakshi1215**
Signed-off-by: Sakshi Patle <sakshi.patle121@gmail.com>

**Description:** 

Adds dockerfile to build ocb docker image, tested in MacOS.

**Link to tracking Issue:**

fixes https://github.com/open-telemetry/opentelemetry-collector/issues/5712

### CODE REVIEW

**Comment by sakshi1215** [comment]
@bogdandrutu @jpkrohling Once this is approved, I will add CI pipelines to build & push the docker images for the same to container registry for the project.

File: cmd/builder/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+COPY go.sum ./
+RUN go mod download
+COPY ./ ./
+
+
+RUN CGO_ENABLED=0 go build \
+    -installsuffix 'static' \
+    -o /app ./main.go
+
+CMD ["/app"]
```
Comment (TylerHelmuth) [inline cmd/builder/Dockerfile:17]:
```suggestion
CMD ["/app"]

```

**Comment by sakshi1215** [comment]
If thats the case, I can make the changes over there repo for the same.

**Review by codeboten — COMMENTED** [review]
Thanks @sakshi1215!

**Comment by jpkrohling** [comment]
I see the -releases repo as the one generating the official distributions. I'm not sure I see the other tools fitting that purpose, like telemetrygen or ocb.

**Comment by codeboten** [comment]
> I see the -releases repo as the one generating the official distributions. I'm not sure I see the other tools fitting that purpose, like telemetrygen or ocb.

I suggested moving to the -releases repo as the collector builder should be released as an official tool no? i see telemetrygen as a development tool rather than something we officially support. That being said I'm not opposed to having tools released from other repos if it makes sense. Personally i find the release of the ocb binary from this repository more confusing than if we said "everything we release is the releases repo".

**Comment by jpkrohling** [comment]
I'm actually liking this idea now. I thought of the releases repo as "official distributions", but "official tools" as you pointed out is an interesting aspect as well.

**Comment by frzifus** [comment]
Is this still a thing? ^^

**Comment by jackgopack4** [comment]
Hey @sakshi1215 or anyone else; did this get dropped in favor of adding docker files to the releases repo? I think there would still be value in having a dockerfile in opentelemetry-collector/cmd/builder and am going to pick this up

**Pull request closed without merging**